databases/backends/postgres.py

```diff
@@ -62,7 +63,9 @@ def _get_connection_kwargs(self) -> dict:
     async def connect(self) -> None:
         assert self._pool is None, "DatabaseBackend is already running"
         kwargs = self._get_connection_kwargs()
-        self._pool = await asyncpg.create_pool(str(self._database_url), **kwargs)
+        self._pool = await asyncpg.create_pool(
+            str(urlparse(self._database_url._url)._replace(query="").geturl()), **kwargs
```

Comment:
Probably something like this instead:

```python
url = str(self._database_url.replace(query=''))
self._pool = await asyncpg.create_pool(url, **kwargs)
```

Reply:
done

Reply:
@tomchristie @euri10 sorry for jumping into, but it'll break other things. The matter is that the query is the only way to pass other params to `asyncpg` like `socket` or `host` that breaks the dsn parsing or caching params, etc.
`self._get_connection_kwargs()` will return only pool size details and the flag for `ssl`, but the rest will erased.
I think we should catch all query params in `self._get_connection_kwargs()` or use an approach used in other backends - 
```
    async def connect(self) -> None:
        assert self._pool is None, "DatabaseBackend is already running"
        kwargs = self._get_connection_kwargs()
        self._pool = await aiomysql.create_pool(
            host=self._database_url.hostname,
            port=self._database_url.port or 3306,
            user=self._database_url.username or getpass.getuser(),
            password=self._database_url.password,
            db=self._database_url.database,
            autocommit=True,
            **kwargs,
        )
```.
The second one seems as a better option as its aligned with the unified approach.

Reply:
does that look better @gvbgduh ?

Reply:
it definitely does @euri10!
But what I missed is that params from query are coming through either.
```
>>> from databases import DatabaseURL
>>> url = DatabaseURL('pg://user:pass@/db?host=host&port=5432')
>>> url.password
'pass'
>>> url.username
'user'
>>> url.database
'db'
>>> url.port
>>> url.hostname
>>> url.options
{'host': 'host', 'port': '5432'}
```
`_get_connection_kwargs()` accesses `url.options`, but looks for specific params, and `DatabaseURL` doesn't look in query for missing things.

It might be reasonable to tweak `DatabaseURL` to look into options/query if it can't find params in the main url, at least for {username, password, database, port, hostname}.

If @tomchristie has no objections

Reply:
it's true that  `_get_connection_kwargs` currently removes anything that is not min_size, max_size or ssl
with an dsn like `f"{POSTGRES_URLS[0]}?min_size=1&max_size=2&sslmode=require"` for instance the sslmode is lost

Reply:
actually, it should pass it as of line https://github.com/encode/databases/blob/master/databases/backends/postgres.py#L58, but those options come from the `Database` instantiation - https://github.com/encode/databases/blob/master/databases/core.py#L32.
So it'll broadcast provided kwargs as is to the DB driver.

Reply:
I can't see the kwarg `sslmode` for the connection or for the pool in `asyncpg`, so you should probably use:
```
ssl – Pass True or an ssl.SSLContext instance to require an SSL connection. If True, a default SSL context returned by ssl.create_default_context() will be used.
```
but in general you should be able to provide it as
```
database = Database('pg://user:pass@/db?host=host&port=5432', server_settings={...}, ...)
```

Reply:
got it, needs to be passed from Database
for reference sslmode is defined [here](https://github.com/MagicStack/asyncpg/blob/ffb9ebffbe7182a4331b03f3fcbf6316ef7b50ea/asyncpg/connect_utils.py#L273) in asyncpg and used in those [tests](https://github.com/MagicStack/asyncpg/blob/2d76f50dccf35cf2f1d70b41ebd6198d2dfff8d7/tests/test_connect.py#L415)

Reply:
yeah, I see, thanks!